### PR TITLE
New Component: DateTimePicker [CAX-774]

### PR DESCRIPTION
**Pull request by elenagiere**
DateTimePicker: combine datePicker and timePicker into one component.

Pending:
- Storybook
- more unit tests

## Description
The selection of a date and time is possible with this component.
There is a plethora of properties to customize this dateTimePicker.

## Motivation and Context
New Component!
Screen Shot 2020-12-09 at 7 54 39 PM

Figma Link: https://www.figma.com/file/wDbhcTS6fYP9wFyJ7VnE0r/Lumos-Components?node-id=11077%3A95346

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/timepicker/TimePicker.ts [hunk 1]
```diff
@@ -291,6 +312,7 @@ export class TimePicker extends LitElement {
     return html `
       <md-combobox
         compact
+        selectWhenInFocus
         class="amPm-combo-box"
```
Comment (futurethang) [inline web-components/src/components/timepicker/TimePicker.ts:316]: Nit: when typing in this box it uses lowercase, but options are uppercase. Maybe `text-transform:uppercase` will do the trick.

**Review by futurethang — CHANGES_REQUESTED** [review]

File: web-components/src/components/combobox/ComboBox.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ export class ComboBox extends FocusMixin(LitElement) {
   @property({ type: Boolean, reflect: true }) ordered = false;
   @property({ type: Boolean, reflect: true }) expanded = false;
   @property({ type: Boolean, reflect: true }) compact = false;
+  @property({ type: Boolean }) selectWhenInFocus = false;
```
Comment (yananym) [inline web-components/src/components/combobox/ComboBox.ts:41]: do we want this to be an attribute? with a lower case - dash naming?

File: web-components/src/components/datepicker/DatePicker.ts [hunk 3]
```diff
@@ -3,19 +3,19 @@ import "@/components/input/Input";
...
-import { customElement, html, internalProperty, LitElement, property, PropertyValues, query } from "lit-element";
+import { customElement, html, internalProperty, LitElement, property, query } from "lit-element";
 import { DateTime } from "luxon";
 
 export namespace DatePicker {}
 export const weekStartDays = ["Sunday", "Monday"];
 @customElement("md-datepicker")
 export class DatePicker extends LitElement {
-  @property({ type: Boolean }) shouldCloseOnSelect = true;
+  @property({ type: Boolean }) shouldCloseOnSelect = false;
```
Comment (yananym) [inline web-components/src/components/datepicker/DatePicker.ts:13]: do we want this to be an attribute also?

File: web-components/src/components/input/Input.ts [hunk 4]
```diff
@@ -162,12 +163,14 @@ export class Input extends FocusMixin(LitElement) {
   @property({ type: Boolean }) isLoading = false;
   @property({ type: Boolean }) compact = false;
   @property({ type: Boolean }) multi = false;
+  @property({ type: Boolean }) selectWhenInFocus = false;
```
Comment (yananym) [inline web-components/src/components/input/Input.ts:166]: and here

File: web-components/src/components/timepicker/TimePicker.ts [hunk 5]
```diff
@@ -60,42 +62,33 @@ export namespace TimePicker {
 
 @customElement("md-timepicker")
 export class TimePicker extends LitElement {
+  @property({ type: Boolean }) twoDigitAutoTab = false;
```
Comment (yananym) [inline web-components/src/components/timepicker/TimePicker.ts:65]: all booleans basically need to also have an attribute friendly name (no camelCase)

File: web-components/src/components/date-time-picker/DateTimePicker.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+@customElement("md-date-time-picker")
+
+export class DateTimePicker extends LitElement {
+  @property({ type: String }) maxDate: string | undefined = undefined;
+  @property({ type: String }) minDate: string | undefined = undefined;
+  @property({ type: String }) dateValue: string | undefined = undefined;
+  @property({ type: String }) weekStart: typeof weekStartDays[number] = "Sunday";
+  @property({ type: String }) locale = "en-US";
+
+  @property({ type: Boolean }) twoDigitAutoTab = false;
```
Comment (yananym) [inline web-components/src/components/date-time-picker/DateTimePicker.ts:24]: attributes please?

**Review by futurethang — DISMISSED** [review]

**Review by yananym — APPROVED** [review]

**Pull request merged**